### DISCUSSION TITLE
Hierarchical HMM with nested step + angle distributions

### DISCUSSION THREAD

**Discussion by picardis** [bmcclintock/momentuHMM#155] [Q&A] [answered]
@salamandermountain and I are trying to fit a hierarchical HMM to segment tracks into nested behavioral states using GPS data at two different temporal resolution. The coarse level uses daily GPS data and is meant to capture transitions between a migratory and a resident state. The fine level uses 2-hour GPS data and is meant to capture transitions between resting, foraging, and traveling behaviors nested within the migratory and resident states. Is it possible to use location data at multiple levels? It seems like prepData() does not allow more than one level in coordLevel. It runs with coordLevel = "1" but not with coordLevel = c("1", "2") or coordLevel = c("1", "2i", "2"). The error it returns is unrelated ("coordLevel must be a character string") but I suspect it's because it only expects location data at one of the levels, while the other level is supposed to have a different type of data stream. Would the current package implementation support using steps and angles as the data streams at both levels?

**Comment by bmcclintock** [comment, marked as answer]
`coordLevel` can only be specified for a single level. The error is related as you have provided a character _vector_, not a character _string_ (see `?prepData`). Regardless, I would suggest that it does not make a lot of sense to duplicate location data at multiple levels. Generally speaking, location data should be included only for the finest level they are available for (if the finer-level movement and/or state transition properties are sufficiently different between the coarser-level states, then the model should be able to identify when coarse-level state switches occur without duplicating the location data). You can of course override this behavior by duplicating the location data streams for the other levels yourself.

**Reply by picardis** [reply]
Thanks, Brett! This is helpful. Do you think it would be possible/make sense to fit a model where the fine-level states within the coarse-level states have the same steps and angles distributions but the transition matrix differs depending on the coarse-level behavioral state? For example, a model with traveling + foraging + resting (fine level) nested within migration + residency (coarse level) where the steps and angles for, say, foraging are the same regardless of whether the animal is migrating or resident, but the transition probability to and from foraging differ depending on whether the animal is migrating or resident?

**Reply by bmcclintock** [reply]
It is certainly possible (see the HHMM examples in the [vignette](https://cran.r-project.org/package=momentuHMM/vignettes/momentuHMM.pdf) and example [code](https://github.com/bmcclintock/momentuHMM/tree/master/vignettes/examples)). I can envision scenarios where this would make sense, but whether or not it makes sense for your particular study system is up to you.
